### ISSUE TITLE
[bug]: AMI confbridge. No final TalkingStatus: off event for user when user leaves confbridge while talking

### ISSUE THREAD

**Issue by wonder75** [asterisk/asterisk#760]
### Severity

Minor

### Versions

20.8.1

### Components/Modules

Confbridge,AMI

### Operating Environment

Debian stable, Asterisk build from source.

### Frequency of Occurrence

Constant

### Issue Description

AMI provides events for the talking status of confbridge participants

```
Event: ConfbridgeTalking
..
TalkingStatus: on

Event: ConfbridgeTalking
..
TalkingStatus: off
```

If a participant leaves a conference while talking, no final TalkingStatus: off event is fired, even though the actual TalkingStatus of the participant for other participants is "off". I encountered this while building a WebRTC application that displayed incorrect talking status because of this. I was able to work around this issue in my code, but it caused quiet a few problems.

As the talking status of a user who has left a conference is actually "off" from the conferences point of view, I am reporting this issue.

### Asterisk Issue Guidelines

- [X] Yes, I have read the Asterisk Issue Guidelines

**Comment by jcolp** [comment]
Is the fact that the channel has left the conference bridge not enough indication that its talking status is no longer applicable/it's no longer able to talk?

I ask because "fixing" this would effectively double the number of events, when I think ConfbridgeLeave should imply this already.

**Comment by wonder75** [comment]
This is correct. ConfbridgeLeave implies TalkingStatus: off, and is for sure a way to work around this problem. But in my opinion the TalkingStatus provided is nevertheless incorrect. As I "trusted" that value initially, I ran into problems, so I decided to report.

**Comment by jcolp** [comment]
I think documentation could be improved here, but I am not comfortable adding a talking event as part of a channel leaving.

**Comment by wonder75** [comment]
Yes, if you define a Talking off event as an event only a participating user can create, this is correct.
If you define it as an event a user creates when he stops talking to the other participants in a conference it's not.

I guess these events are used mainly in applications to provide the talking status to other users in a GUI, so it really comes down to a philosophical question if this is a bug or not.
